.github/workflows/build.yml

```diff
@@ -0,0 +1,170 @@
+          python -m pip install 'build'
+
+      - name: Build sdist tarball
+        shell: bash
+        run: |
+          if [[ ! -z "$OVERRIDE_VERSION" ]]; then echo "$OVERRIDE_VERSION" > VERSION; fi
+          # The build package is the reference PEP 517 package builder.  All
+          # dependencies are specified by our setup code.
+          python -m build --sdist .
+
+      # Zip files are not part of PEP 517, so we need to make our own.
+      - name: Create zipfile from tarball
+        shell: bash
+        working-directory: dist
+        run: |
+          # First assert that there is exactly one tarball, and find its name.
+          shopt -s failglob
+          tarball_pattern="*.tar.gz"
+          tarballs=($tarball_pattern)
+          [[ ${#tarballs[@]} == 1 ]]
+          tarball="${tarballs[0]}"
+          # Get the stem and make the zipfile name.
+          stem="${tarball%.tar.gz}"
+          zipfile="${stem}.zip"
+          # Extract the tarball and rezip it.
+          tar -xzf "$tarball"
+          zip "$zipfile" -r "$stem"
+          rm -r "$stem"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: sdist
+          path: |
+            dist/*.tar.gz
+            dist/*.zip
+          if-no-files-found: error
+
+
+  build_wheels:
+    name: Build wheels
+    needs: deploy_test
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_VERSION: ${{ github.event.inputs.override_version }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          # This is about the build environment, not the released wheel version.
+          python-version: '3.7'
+
+      - name: Install pip build
+        run: |
+          python -m pip install 'build'
+
+      - name: Build wheels
+        shell: bash
+        run: |
+          # If the version override was specified, then write it the VERSION
+          # file with it.
+          if [[ ! -z "$OVERRIDE_VERSION" ]]; then echo "$OVERRIDE_VERSION" > VERSION; fi
+          # The build package is the reference PEP 517 package builder.  All
+          # dependencies are specified by our setup code.
+          python -m build --wheel --outdir wheelhouse .
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: ./wheelhouse/*.whl
+
+
+  deploy:
+    name: "Deploy to PyPI if desired"
+    # The confirmation is tested explicitly in `deploy_test`, so we know it is
+    # either a missing confirmation (so we shouldn't run this job) or a valid
+    # confirmation.  We don't need to retest the value of the confirmation,
+    # beyond checking that one existed.
+    if: ${{ github.event.inputs.confirm_ref != '' }}
+    needs: [deploy_test, build_sdist, build_wheels]
+    runs-on: ubuntu-latest
+    env:
+      TWINE_USERNAME: __token__
+      TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      TWINE_NON_INTERACTIVE: 1
+      TWINE_REPOSITORY: pypi
```

Comment:
Oh, a comment here - I see in the settings of this repo that you've called the token `PYPI_API_TOKEN`, which is a mismatch with this line.

Also, we should probably move the env specification on to the "Upload sdist and wheels to PyPI" _step_, not the entire job, just for the purposes of really minimising the time that our secrets are available in the environment.  It's not a big deal at all, since only QuTiP maintainers can run this workflow, but still, just seems like the right thing to do (similar on qutip/qutip).